lms/static/js/student_profile/views/learner_profile_factory.js

```diff
@@ -102,20 +100,38 @@
                 learnerProfileView.showLoadingError();
             };
 
-            var renderLearnerProfileView = function() {
+            var getProfileVisibility = function() {
+                if (options.has_preferences_access) {
+                    return accountPreferencesModel.get('account_privacy');
+                } else {
+                    return accountSettingsModel.get('profile_is_public') ? 'all_users' : 'private';
+                }
+            };
+
+            var showLearnerProfileView = function() {
+                // Record that the profile page was viewed
+                Logger.log('edx.user.settings.viewed', {
+                    page: "profile",
+                    visibility: getProfileVisibility(),
+                    user_id: options.profile_user_id,
+                    requires_parental_consent: options.profile_requires_parental_consent
```

Comment:
@cahrens @symbolist @explorerleslie: note that it was tricky to record the parental consent state when viewing a different user's profile. The APIs do not return this information (presumably it is PII) but in order to do client-side eventing I had to write the information into the page. Let me know if you consider this an issue that anyone can find out if someone is under thirteen just by visiting their profile page and then inspecting the page source.